api/deploy.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"context"
+)
+
+// DeploymentConfig is the API message for deployment configurations.
+type DeploymentConfig struct {
+	ID int `jsonapi:"primary,deploymentConfig"`
+
+	// Standard fields
+	CreatorID int
+	Creator   *Principal `jsonapi:"attr,creator"`
+	CreatedTs int64      `jsonapi:"attr,createdTs"`
+	UpdaterID int
+	Updater   *Principal `jsonapi:"attr,updater"`
+	UpdatedTs int64      `jsonapi:"attr,updatedTs"`
+
+	// Related fields
+	ProjectID int
+	Project   *Project `jsonapi:"relation,project"`
+
+	// Domain specific fields
+	Name string `jsonapi:"attr,name"`
+	// Payload encapsulates DeploymentSchedule in json string format. We use json instead jsonapi because this configuration isn't queryable as HTTP format.
+	Payload string `jsonapi:"attr,payload"`
+}
+
+// DeploymentSchedule is the API message for deployment schedule.
+type DeploymentSchedule struct {
+	Deployments []*Deployment `json:"deployments"`
+}
+
+// Deployment is the API message for deployment.
+type Deployment struct {
+	Spec *DeploymentSpec `json:"spec"`
+}
+
+// DeploymentSpec is the API message for deployment specification.
+type DeploymentSpec struct {
+	Selector *LabelSelector `json:"selector"`
+}
+
+// LabelSelector is the API message for label selector.
+type LabelSelector struct {
+	// MatchExpressions is a list of label selector requirements. The requirements are ANDed.
+	MatchExpressions *LabelSelectorRequirement `json:"matchExpressions"`
+}
+
+// OperatorType is the type of label selector requirement operator.
+type OperatorType string
+
+const (
+	// InOperatorType is the operator type for In.
+	InOperatorType OperatorType = "In"
+	// ExistsOperatorType is the operator type for Exists.
+	ExistsOperatorType OperatorType = "Exists"
+)
+
+// LabelSelectorRequirement is the API message for label selector.
+type LabelSelectorRequirement struct {
+	// Key is the label key that the selector applies to.
+	Key string `json:"key"`
+
+	// Operator represents a key's relationship to a set of values. Valid operators are In, Exists.
+	// Note: NotIn and DoesNotExist are not supported initially.
```

Comment:
"Valid operators are In, Exists.
	// Note: NotIn and DoesNotExist are not supported initially."
	
This could be moved to the "type OperatorType string" place